script.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const tableContainer = document.createElement('div');
+tableContainer.className = 'table-container';
+document.querySelector('body').append(tableContainer);
+
+const table = document.createElement('table');
+tableContainer.append(table);
+table.className = 'todo-list-table';
+
+const thead = document.createElement('thead');
+const tbody = document.createElement('tbody');
+table.append(thead, tbody);
+
+const theadItems = ['To Do Item', 'Name', 'Status'];
+
+for(let i = 0; i < theadItems.length; i++){
+  const theadItem = document.createElement('th');
+  theadItem.textContent = theadItems[i];
+  thead.append(theadItem);
+}
+
+function setNameCell(user, usersInfoArray) {
+  const cellName = document.createElement('td');
+  const userName = usersInfoArray.find(person => person.id === user.userId).name;
+  const emailAdd = usersInfoArray.find(person => person.id === user.userId).email;
```

Comment:
do not repeat find, use destruction
Also, you may create usersHashMap to access users in constant time